### PR TITLE
Fix deadlock on destroy

### PR DESCRIPTION
**Pull request by crimsonred**

### CODE REVIEW

File: .pubnub.yml [hunk 1]
```diff
@@ -2,6 +2,9 @@
 changelog: 
   - 
     changes:
```
Comment (davidnub) [inline .pubnub.yml]: Don't we need to update the `version: ` ?

File: pubnub.go [hunk 2]
```diff
@@ -338,11 +338,14 @@ func (pn *PubNub) DeleteMessagesWithContext(ctx Context) *historyDeleteBuilder {
...
 func (pn *PubNub) Destroy() {
+	close(pn.jobQueue)
 	pn.Config.Log.Println("Calling Destroy")
 	pn.cancel()
-	pn.Config.Log.Println("calling RemoveAllListeners")
-	pn.subscriptionManager.RemoveAllListeners()
-	pn.Config.Log.Println("after RemoveAllListeners")
+	if pn.subscriptionManager != nil {
+		pn.subscriptionManager.Destroy()
+		pn.Config.Log.Println("after sm Destroy")
```
Comment (davidnub) [inline pubnub.go]: Can we spell out `sm` => `subscription manager`?

File: pubnub.go [hunk 3]
```diff
@@ -352,10 +355,16 @@ func (pn *PubNub) Destroy() {
 		pn.Config.Log.Println("after exitTelemetryManager")
 	}
 	pn.Config.Log.Println("calling subscriptionManager Destroy")
-	pn.heartbeatManager.Destroy()
-	pn.subscriptionManager.Destroy()
+	if pn.heartbeatManager != nil {
+		pn.heartbeatManager.Destroy()
+		pn.Config.Log.Println("after hb Destroy")
```
Comment (davidnub) [inline pubnub.go]: Can we spell out `hb` => `heartbeat manager` ?

File: .pubnub.yml (hunk 1 above)
Comment (crimsonred) [reply .pubnub.yml]: version was updated last time for the uuid changes, adding this change in the same version as 
those changes was not merged to master

Comment (davidnub) [reply .pubnub.yml]: oh ok thanks

**Pull request merged**